packages/driver/src/cy/commands/cookies.ts

```diff
@@ -116,27 +142,39 @@ export default function (Commands, Cypress, cy, state, config) {
       return $errUtils.throwErrByPath('cookies.timed_out', {
         onFail: log,
         args: {
-          cmd: getCommandFromEvent(event),
+          cmd: commandName,
           timeout,
         },
       })
     })
   }
 
-  const getAndClear = (log?, timeout?, options = {}) => {
-    return automateCookies('get:cookies', options, log, timeout)
-    .then((resp) => {
+  const getAndClear = ({ commandName, log, options, timeout }: GetAndClearOptions) => {
+    return automateCookies({
+      event: 'get:cookies',
+      commandName,
+      options,
+      log,
+      timeout,
+    })
+    .then((cookies: Cypress.Cookie[]) => {
```

Comment:
Why does the backend need the list of cookies to clear them all out? If no data is passed when the automate event is `clear:cookies` why not delete them all?

I've always wondered that with the session's cookies logic.

Reply:
For the CDP, you can either clear all cookies or clear certain cookies by name, so I guess the original idea was to get the cookies we want to clear, iterate over them, and do it by name (instead of what we _have_ to do with webkit where we clear all cookies then put back ones we didn't want to clear). Also, it keeps parity with `getCookies` since whatever it returns should be the ones cleared.

For `clearAllCookies`, I could use [Network.clearBrowserCookies](https://chromedevtools.github.io/devtools-protocol/tot/Network/#method-clearBrowserCookies) but I figured it wasn't worth having a completely different code path for it since the plumbing already exists.